header/helper.go

```diff
@@ -0,0 +1,36 @@
+package header
+
+import (
+	"bytes"
+	"sort"
+)
+
+// GetFrequentHeader chooses the most frequent header.
+// NOTE: if slice does not have duplicates, then GetFrequentHeader returns the first entry of the passed slice.
+func GetFrequentHeader(headers []*ExtendedHeader) *ExtendedHeader {
+	// sort in the decreasing order
+	sort.Slice(headers, func(i, j int) bool {
```

Comment:
Does the array need to be sortted? The method says to get the most frequent header, does it also need to be the most recent? If so the method name and docstring should be updated. 

If it order doesn't matter, you could have also simplified this algorithm by just filling up a map of `hash` to `int` where the `int` is the frequency counter. Then you could just need the `maxCount` variable to keep track fo the most frequent.